docs/integrations/webhooks/inngest.mdx

```diff
@@ -0,0 +1,215 @@
+     e.id === user.primary_email_address_id
+   ).email
+   await database.users.insert({ id, email, first_name, last_name })
+ }
+)
+```
+
+
+The `event` object contains all of the relevant data for the event. The `event.data` will match the `data` object from the standard Clerk webhook [payload structure](/docs/integrations/webhooks#payload-structure). With this `clerk/user.created` event, the `event.data` will be a Clerk User json object.
+
+As you can see, you can choose which events you want to handle with each function. You might write a separate function for `clerk/user.updated` and `clerk/user.deleted` handling the entire lifecycle end to end.
+
+Note that multiple functions can also listen to the same event. This pattern is called “[fan-out](https://www.inngest.com/docs/guides/fan-out-jobs).”
+
+
+## Creating a function to send a welcome email
+
+Often, applications need to perform additional tasks when a new user is created, like send a welcome email with tips and useful information.
+
+While it is possible to add this logic at the end of your sync function as seen in the [previous section](/docs/integrations/webhooks/inngest#creating-a-function-to-sync-a-new-user-to-a-database), it's better to decouple logic into different functions so issues with one task do not affect the other ones. For example, if your email fails to send, it should not affect adding the user to an email list.
+
+With Inngest, each function has automatic retries, so only the code that has issues is re-run.
+
+The code below creates another function using the same `clerk/user.created` event and adds the logic to send the welcome email:
+
+
+```typescript
+const sendWelcomeEmail = inngest.createFunction(
+ { id: 'send-welcome-email' },
+ { event: 'clerk/user.created' },
+ async ({ event }) => {
+   const { user } = event.data
+   const { first_name } = user
+   const email = user.email_addresses.find(e =>
+     e.id === user.primary_email_address_id
+   ).email
+   await emails.sendWelcomeEmail({ email, first_name })
+ }
+)
+```
+
+
+Now, you have a function that utilizes the same Clerk webhook event for another purpose. Clerk webhook events can be used for all sorts of application lifecycle use cases. For example, adding users to a marketing email list, starting a Stripe trial, or provisioning new account resources.
+
+
+### Sending a delayed follow-up email
+
+Every Inngest function handler has an additional `step` object which provides tools to create more complex functions. Using `step.run` allows you to encapsulate specific code that will be automatically retried ensuring that issues with one part of your function don't force the entire function to re-run. Additionally, [other tools like `step.sleep`](https://www.inngest.com/docs/reference/functions/step-sleep), are available to extend functionality.
+
+The code below sends a welcome email, then uses `step.sleep` to wait for three days before sending another email offering a free trial:
```

Comment:
In the previous section the guide indicates to decouple tasks so they don't affect each other, but this example has two tasks coupled together. Is there a reason for this? It came off weird reading the guide going from the previous section to this one section.

Reply:
That’s a good point. It makes sense to decouple either big tasks or unrelated tasks. Here, one task results from another and they constitute one whole.

I'll rework the previous section a little to make this clearer!

Reply:
See https://github.com/clerk/clerk-docs/pull/645/files/388815e0218521951e3e4dc48ebc67989798c6a9#r1459451471